### PR TITLE
PLAT-632: use the new deploy-service which uses the github oidc provider

### PR DESCRIPTION
**Pull request by bjarneo**
* Remove AWS and Kube config from secrets input
* Use role to assume, with a role coming from the repository using this workflow
* Add the logic for the dynamic kube config file to the teardown
* Updated the dockerfile to use nonroot user. (also using serve instead of nginx)

### CODE REVIEW

File: .github/workflows/deploy-preview.yaml [hunk 1]
```diff
@@ -133,22 +137,19 @@ jobs:
             ${{ env.ECR_REGISTRY }}/${{ env.ECR_REPOSITORY }}:${{ env.INSTANCE }}-${{ inputs.app_name_postfix }}-${{ steps.app_name.outputs.full_name }}-${{ steps.short_id.outputs.sha }}
 
       - name: Deploy the service to Kubernetes
-        uses: dignio/deploy-service@v2
+        uses: dignio/deploy-service@v3
         with:
           app_name: ${{ steps.app_name.outputs.full_name }}
           service_type: ${{ inputs.service_type }}
+          cluster_name: ${{ env.CLUSTER_NAME }}
+          instance: ${{ env.ISNTANCE }}
```
Comment (weel) [inline .github/workflows/deploy-preview.yaml:145]: Typo

File: README.md [hunk 2]
```diff
@@ -21,15 +21,13 @@ jobs:
       app_name: prevent-ui
       app_name_postfix: storybook
       service_type: webservice
+      aws_role: arn:aws:iam::<org_id>:role/github_actions_kubernetes_deployment_development
```
Comment (weel) [inline README.md:24]: Nitpick, this is the AWS Account ID

**Review by weel — COMMENTED** [review]
Fix typo, and should be good to go.

**Comment by bjarneo** [comment]
Thanks, and, fixed, @weel

**Review by weel — APPROVED** [review]
LGTM 👍

**Pull request merged**